hal/halx86/smp/i386/apspinup.S

```diff
@@ -0,0 +1,77 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Assembly file for protected mode AP code
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+#include <asm.inc>
+#include <ks386.inc>
+
+PUBLIC _APSpinup
+PUBLIC _APSpinupEnd
+PUBLIC _APProcessorStateStruct
+EXTERN _APEntry:DWORD
+
+.code32
+_APSpinup:
+
+    /* At this point the value in ebx is the location of the bootstub + the size of the APEntry segment */
+    mov ecx, ebx
+    add ecx, [_APSpinupEnd - _APSpinup]
+    /* Load a 4kb stack for AP */
+    mov esp, [ecx + HEX(4096)] 
+
+    /* Load the GDT based on BSP */
+
+    /* TODO: replace this with something that's going to actually work */
+    xor eax, eax
+    mov eax, ([ebx] + [APCr3 - _APSpinup])
+    mov cr3, eax
+
+    /* Load some ProcessorState values into proper area */
```

Comment:
The full ProcessorState must be loaded here (control, debug, segment, general purpose registers, etc.)

Reply:
You need to prepare an AP for calling `KiSystemStartup`. In this particular case, I'd say GP registers are needed to be zeroed it least

Reply:
But that has nothing to do with processorstate, also don't i need to push a parameter on the stack? that already removes atleast 1 one from being zeroed before this occurs

Reply:
ProcessorState contains the whole processor state, that's what it has to do :)
You can just be dumb here and load what it has unconditionally, and let KeStartAllProcessors decide what should be put there. You may suggest a different solution, but I feel like that would be a good separation of responsibility areas